stress_test_model_loan_book.R

```diff
@@ -530,9 +523,29 @@ for (i in seq(1, nrow(transition_scenarios))) {
       )
   }
 
+  loanbook_annual_profits <- loanbook_annual_profits %>%
+    # ADO 879: removed company id from join, but should be re-introduced later on
+    left_join(financial_data_loans, by = c("company_name", "ald_sector", "technology", "year"))
```

Comment:
what is the reasoning behind using a left join here, and what is the hanlding for the resulting NAs on e.j. column pd?

Reply:
This is a good example where there technically is not need to keep the NA rows. But if we drop them, we may want to think about adding them to some json output column/file that records what is kicked out where and why...

Reply:
how about as a short term solution dropping them (because keeping them just adds complication and makes understanding things harder) and adding a log line (simply use cat) how many rows where dropped. For proper handling i.e. returning affected cases in some form a follow up ticket can (should) be created)

Reply:
sounds good